src/Http/Routing/src/Template/RouteTemplate.cs

```diff
@@ -28,7 +28,14 @@ public RouteTemplate(RoutePattern other)
         // RequiredValues will be ignored. RouteTemplate doesn't support them.
 
         TemplateText = other.RawText;
-        Segments = new List<TemplateSegment>(other.PathSegments.Select(p => new TemplateSegment(p)));
+
+        var segments = new List<TemplateSegment>(other.PathSegments.Count);
```

Comment:
nit: why not assign directly to `Segments`?

Reply:
I didn't because `Segments` is an `IList`. Doing it directly against the List itself eliminates the virtual calls to `Add`. But could change that one line if you felt strongly. It probably doesn't matter either way.

Reply:
Fair enough, just not obvious as a reviewer unless I pulled down the code, or looked very very closely 😃

Reply:
@eerhardt Add a comment to make it obvious to future readers

Reply:
I ended up just assigning directly to `Segments`. It felt inconsistent with `Parameters` below, and making more changes here didn't feel right.

Reply:
I ended up just assigning directly to `Segments`. It felt inconsistent with `Parameters` below, and making more changes here didn't feel right.